exir/program/_program.py

```diff
@@ -1439,20 +1442,29 @@ def transform(
 
     @et_logger("to_backend")
     def to_backend(
-        self, partitioner: Union[Partitioner, Dict[str, Partitioner]]
+        self,
+        partitioner: Union[
+            DelegationSpec,
```

Comment:
why not just ask users to wrap their delegate spec in the `AllNodePartitioner` and make this cleaner API?